### PR TITLE
Make HTTP server set headers for not caching files. Fixes #215

### LINKED ISSUE #215: Don't serve up 304s for static content
If a user switches from a new version of tnoodle to an old version and then back to the new version, we still want to serve up the full contents of our static HTML/CSS/JS everytime, without regard to the modification times of the files on our filesystem.

2016-06-09_15 18 13_1111x364_kaladin

### PR DESCRIPTION
**Pull request by gyroninja**
Appends to the header that winstone uses for it's internal implementation of http server.

### CODE REVIEW

**Comment by gyroninja** [comment]
(I'm also assuming this project is meant to be compatible with Java 6)

**Comment by jfly** [comment]
Hmmm... I see the following behavior with your changes.

```
~/gitting/tnoodle @kaladin> curl -I http://localhost:2014/scrambler-interface/js/ui.js
HTTP/1.1 200 OK
Server: Winstone Servlet Engine v0.9.10
Cache-Control: no-cache, no-store, must-revalidate
Pragma: no-cache
Expires: 0
Content-Type: text/javascript;charset=ISO-8859-1
Content-Length: 70417
Last-Modified: Wed, 31 Aug 2016 03:12:19 GMT
Connection: Keep-Alive
Date: Wed, 31 Aug 2016 03:25:12 GMT
X-Powered-By: Servlet/2.5 (Winstone/0.9.10)

~/gitting/tnoodle @kaladin> curl -I http://localhost:2014/scrambler-interface/js/ui.js -H "If-Modified-Since: Wed, 31 Aug 2016 03:12:19 GMT"
HTTP/1.1 304 Not Modified
Server: Winstone Servlet Engine v0.9.10
Cache-Control: no-cache, no-store, must-revalidate
Pragma: no-cache
Expires: 0
Content-Type: text/javascript;charset=ISO-8859-1
Content-Length: 0
Connection: Keep-Alive
Date: Wed, 31 Aug 2016 03:25:25 GMT
X-Powered-By: Servlet/2.5 (Winstone/0.9.10)
`````

Note how that in the second request we sent the `If-Modified-Since` header, and the server responded with a HTTP 304. I believe the headers you added should teach web browsers to never send the `If-Modified-Since` header, but we have this problem that web browsers that have communicated with an old version of TNoodle won't know that they should not send a `If-Modified-Since`, and I believe they'll behave like this second curl request and end up with a stale version of `ui.js`.

It looks like Winstone has 2 places where it issues 304 responses: [here](https://github.com/cubing/tnoodle/blob/2ea72bcabbe15b614a19d8eed9d2cbe5652be89f/winstone/src/winstone/StaticResourceServlet.java#L134-L144) and [here](https://github.com/cubing/tnoodle/blob/2ea72bcabbe15b614a19d8eed9d2cbe5652be89f/winstone/src/javax/servlet/http/HttpServlet.java#L106-L113). I'm pretty sure the first one is the code branch we're hitting here. I actually wouldn't be opposed to modifying Winstone and just commenting out this code. If that sounds too gross, a cleaner solution might be to write some sort of servlet middleware that removes the `If-Modified-Since` header before it gets to this part of Winstone.

File: winstone/src/javax/servlet/http/HttpServlet.java [hunk 1]
```diff
@@ -103,14 +102,14 @@ protected void service(HttpServletRequest request,
             if (lastModified == -1)
                 doGet(request, response);
             else {
-                long ifModifiedSince = request.getDateHeader(HEADER_IFMODSINCE);
-                if (ifModifiedSince < (lastModified / 1000 * 1000)) {
-                    if (!response.containsHeader(HEADER_LASTMOD)
-                            && (lastModified >= 0))
-                        response.setDateHeader(HEADER_LASTMOD, lastModified);
+//                long ifModifiedSince = request.getDateHeader(HEADER_IFMODSINCE);
```
Comment (jfly) [inline winstone/src/javax/servlet/http/HttpServlet.java]: Could you add a comment here that links to https://github.com/cubing/tnoodle/issues/215?

File: winstone/src/winstone/StaticResourceServlet.java [hunk 2]
```diff
@@ -131,17 +130,17 @@ else if (res.isDirectory()) {
...
-                && (cachedResDate >= (res.lastModified() / 1000L * 1000L))) {
-            String mimeType = getServletContext().getMimeType(
-                    res.getName().toLowerCase());
-            if (mimeType != null)
-                response.setContentType(mimeType);
-            response.setStatus(HttpServletResponse.SC_NOT_MODIFIED);
-            response.setContentLength(0);
-            response.flushBuffer();
-        }
+//        else if (!isInclude && (cachedResDate != -1)
```
Comment (jfly) [inline winstone/src/winstone/StaticResourceServlet.java]: As above, could you add a comment here that links to https://github.com/cubing/tnoodle/issues/215?

**Comment by jfly** [comment]
Awesome, this looks like it's working now!

```
~/gitting/tnoodle @kaladin> curl -I localhost:2014/scrambler-interface/js/ui.js
HTTP/1.1 200 OK
Server: Winstone Servlet Engine v0.9.10
Cache-Control: no-cache, no-store, must-revalidate
Pragma: no-cache
Expires: 0
Content-Type: text/javascript;charset=ISO-8859-1
Content-Length: 70417
Last-Modified: Wed, 31 Aug 2016 05:21:49 GMT
Connection: Keep-Alive
Date: Wed, 31 Aug 2016 05:22:10 GMT
X-Powered-By: Servlet/2.5 (Winstone/0.9.10)

~/gitting/tnoodle @kaladin> curl -I localhost:2014/scrambler-interface/js/ui.js -H "If-Modified-Since: Wed, 31 Aug 2016 05:21:49 GMT"
HTTP/1.1 200 OK
Server: Winstone Servlet Engine v0.9.10
Cache-Control: no-cache, no-store, must-revalidate
Pragma: no-cache
Expires: 0
Content-Type: text/javascript;charset=ISO-8859-1
Content-Length: 70417
Last-Modified: Wed, 31 Aug 2016 05:21:49 GMT
Connection: Keep-Alive
Date: Wed, 31 Aug 2016 05:22:17 GMT
X-Powered-By: Servlet/2.5 (Winstone/0.9.10)


```

**Comment by jfly** [comment]
LGTM!

**Pull request merged**